batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,164 @@
+    ) {
+
+      ps.setString(1, student.getName());
+      ps.setString(2, student.getAddress());
+      ps.setDate(3, new java.sql.Date(student.getDob().getTime()));
+      ps.setString(4, student.getDepartment());
+      ps.setString(5, student.getBatch());
+      ps.setInt(6, student.getRoll());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void delete(Integer studentId) throws DataException {
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.DELETE_FROM_STUDENT)
+    ) {
+
+      ps.setInt(1, studentId);
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public void update(Student student) throws DataException {
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.UPDATE_STUDENT)
+    ) {
+      ps.setString(1, student.getName());
+      ps.setString(2, student.getAddress());
+      ps.setDate(3, new java.sql.Date(student.getDob().getTime()));
+      ps.setString(4, student.getDepartment());
+      ps.setString(5, student.getBatch());
+      ps.setInt(6, student.getRoll());
+      ps.setInt(7, student.getId());
+      ps.executeUpdate();
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Student fetchById(Integer id) throws DataException {
+    ResultSet studentResult;
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.FETCH_STUDENT_BY_ID)
+    ) {
+      ps.setInt(1, id);
+      studentResult = ps.executeQuery();
+
+      Student student = null;
+      if (studentResult.next()) {
+        student = new Student();
+        student.setId(studentResult.getInt(EntityConstant.ID));
+        student.setRoll(studentResult.getInt(EntityConstant.ROLL));
+        student.setDepartment(studentResult.getString(EntityConstant.DEPARTMENT));
+        student.setBatch(studentResult.getString(EntityConstant.BATCH));
+
+        student.setName(studentResult.getString(EntityConstant.NAME));
+        student.setAddress(studentResult.getString(EntityConstant.ADDRESS));
+        student.setDob(studentResult.getDate(EntityConstant.DOB));
+
+      }
+      return student;
+
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+  }
+
+  public Integer fetchTotalRecordNumber() throws DataException {
+    ResultSet rs;
+    try (Connection conn = DbUtils.getConnection();
+            PreparedStatement ps = conn.prepareStatement(QueryConstant.FETCH_TOTAL_STUDENT_COUNT)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)